### ISSUE TITLE
Evaluating a Function at a given point

### ISSUE THREAD

**Issue by ctjacobs** [firedrakeproject/firedrake#155]
I'm trying to evaluate a Function at a given point in space using:

```
from firedrake import *

mesh = UnitSquareMesh(10, 10)
fs = FunctionSpace(mesh, "CG", 1)

f = Function(fs).interpolate(Expression("cos(x[0])"))

xy = (1.0, 1.0)
print f(xy)
```

(also see the DOLFIN example [here](http://fenicsproject.org/documentation/tutorial/fundamentals.html#examining-the-discrete-solution)), and I get this error message:

```
Expecting dim to match the geometric dimension, got dim=0 and gdim=2.
Couldn't map 'function_1002' to a float, returning ufl object without evaluation.
function_1002
```

**Comment by miklos1** [comment]
@ctjacobs: Do you still need this?

**Comment by ctjacobs** [comment]
Yes.

**Comment by miklos1** [comment]
Fixed in #612, closing.
